src/commands/heroku.rs

```diff
@@ -414,23 +415,76 @@ pub fn deploy_app(ctx: &mut Context, msg: &Message, mut args: Args) -> CommandRe
 
     msg.reply(&ctx, build_response(&app_name, &build))?;
 
-    let mut build_pending = true;
+    let build_pending = Arc::new(Mutex::new(true));
+    let build_pending_value = Arc::clone(&build_pending);
+
+    let build = heroku_client(ctx).request(&builds::BuildDetails {
+        app_id: app_name.clone(),
+        build_id: build.clone().id,
+    })?;
+
+    let build_status = Arc::new(Mutex::new(build.clone().status));
+    let build_status_value = Arc::clone(&build_status);
 
-    while build_pending {
+    let mut sched = JobScheduler::new();
+
+    let build_check_interval = bot_config(&ctx).build_check_interval.clone();
+
+    // Set up job to periodically check if the build is complete
+    sched.add(Job::new(
+        job_interval(build_check_interval).parse().unwrap(),
+        || {
+            if *build_status.lock().unwrap() != "pending" {
+                let mut value = build_pending_value.lock().unwrap();
+                *value = false;
+            }
+        },
+    ));
+
+    // Set up job to periodically display a build status message in Discord
+    let build_id = build.clone().id;
+    let context = ctx.clone();
+    let build_message_display_interval = bot_config(&ctx).build_message_display_interval.clone();
+
+    sched.add(Job::new(
+        job_interval(build_message_display_interval)
+            .parse()
+            .unwrap(),
+        move || {
+
+            // Doing manual error handling
+            // Because the try (?) operator cannot be 
+            // used in a closure
+            // (As of July 2020)
+            let result = msg.channel_id
+                .say(&context, format!("Build {} is still pending...", build_id));
+
+            // Printing to the console as 
+            // An error cannot be propogated from an closure
+            // up to the 
+            // enclosing function
+            // (As of July 2020)
+            match result {
+                Ok(_result) => {},
+                Err(e) => {
+                    println!("An error occured when trying to post the build pending message for {}: {}", build_id, e);
+                }
+            }
+        },
+    ));
+
+    while *build_pending.lock().unwrap() {
         let build = heroku_client(ctx).request(&builds::BuildDetails {
             app_id: app_name.clone(),
             build_id: build.clone().id,
         })?;
```

Comment:
Did you mean to make this API call in the `build_check` scheduled job? This way you're calling Heroku two times a second.